### PR TITLE
[Season 2][ABO-144 | ABO-145] Logic adjustments

### PR DESCRIPTION
**Pull request by DidacSF**
## Description

This PR applies the changes mentioned in the following JIRA tickets:

* https://ajunanetwork.atlassian.net/browse/ABO-144
* https://ajunanetwork.atlassian.net/browse/ABO-145

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [ ] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [x] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [x] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [x] Tests for the changes have been added
- [x] Necessary documentation is added (if appropriate)
- [x] Formatted with `cargo fmt --all`
- [x] Linted with `cargo clippy --all-features --all-targets`
- [x] Tested with `cargo test --workspace --all-features --all-targets`

### CODE REVIEW

**Review by darkfriend77 — COMMENTED** [review]
check comments

File: pallets/ajuna-awesome-avatars/src/types/avatar/versions/v2/mod.rs [hunk 1]
```diff
@@ -234,14 +234,16 @@ impl<T: Config> ForgerV2<T> {
 					if sacrifices.iter().all(|sacrifice| {
 						let equippable_item = sacrifice.get_item_sub_type::<EquippableItemType>();
 						sacrifice.has_type(ItemType::Equippable) &&
-							sacrifice.same_rarity(leader) &&
+							sacrifice.get_rarity() == RarityTier::Legendary &&
 							sacrifice.same_class_type_2(leader) &&
 							(equippable_item.is_armor_base() || equippable_item.is_weapon())
-					}) {
+					}) && leader.get_rarity() >= RarityTier::Legendary
```
[the remark this answers is not in the document]
Comment (DidacSF) [reply pallets/ajuna-awesome-avatars/src/types/avatar/versions/v2/mod.rs:240]: Yeah, this is more a defensive check, I think it does no damage to keep here

File: pallets/ajuna-awesome-avatars/src/types/avatar/versions/v2/mod.rs [hunk 2]
```diff
@@ -1218,14 +1274,14 @@ mod test {
...
+			assert_eq!(forge_type_map.get(&ForgeType::None).unwrap(), &49207);
 			assert_eq!(forge_type_map.get(&ForgeType::Stack).unwrap(), &26687);
 			assert_eq!(forge_type_map.get(&ForgeType::Tinker).unwrap(), &2407);
 			assert_eq!(forge_type_map.get(&ForgeType::Build).unwrap(), &4723);
 			assert_eq!(forge_type_map.get(&ForgeType::Assemble).unwrap(), &5620);
 			assert_eq!(forge_type_map.get(&ForgeType::Breed).unwrap(), &1558);
-			assert_eq!(forge_type_map.get(&ForgeType::Equip).unwrap(), &848);
-			assert_eq!(forge_type_map.get(&ForgeType::Mate).unwrap(), &2122);
+			assert_eq!(forge_type_map.get(&ForgeType::Equip).unwrap(), &94);
+			assert_eq!(forge_type_map.get(&ForgeType::Mate).unwrap(), &1872);
```
[the remark this answers is not in the document]
Comment (DidacSF) [reply pallets/ajuna-awesome-avatars/src/types/avatar/versions/v2/mod.rs:1284]: They way we generate the avatars in the test, makes it that the Rarity is randomized, so adding the additional checks for both Mate and Equip has made that some avatars that would have ended up there are now outside the checks, that I think explains the decrease specially in equip that didn't have the rarity restrictions before.

**Review by darkfriend77 — APPROVED** [review]
good

**Pull request merged**